kratos/utilities/properties_accessor.h

```diff
@@ -0,0 +1,322 @@
+                                                const Vector&,
+                                                const int)>;
+
+    using DoubleConfiguration = std::map<DoubleVariableType, DoubleFunction>;
+
+    // std::function<void(int)>
+
+    // using std::unordered_map<DoubleVariableType>
+
+    ///@}
+    ///@name Life Cycle
+    ///@{
+
+    /// Default constructor.
+    PropertiesAccessor(Properties* pParentProperties)
+        : mpParentProperties(pParentProperties) {}
+
+    /// Destructor.
+    virtual ~PropertiesAccessor() {}
+
+
+    ///@}
+    ///@name Operators
+    ///@{
+
+
+    ///@}
+    ///@name Operations
+    ///@{
+
+    template<class TVariableType>
+    typename TVariableType::Type& GetValue(const TVariableType& rV,
+                                           const GeometryType& rpGeometry,
+                                           const DataValueContainer& rDataContainer,
+                                           const ProcessInfo& rCurrentProcessInfo,
+                                           const Vector& rShapeFunctionValues,
+                                           const int GaussPointIndex=0)
+    {
+        mpParentProperties->GetValue(rV);
+    }
+
+
+    double GetValue(const DoubleVariableType& rV,
+                     const GeometryType& rGeometry,
+                     const DataValueContainer& rDataContainer,
+                     const ProcessInfo& rCurrentProcessInfo,
+                     const Vector& rShapeFunctionValues,
+                     const int GaussPointIndex=0)
+    {
+        if(mConfigurationsDouble.find(rV) != mConfigurationsDouble.end())
+            return mConfigurationsDouble[rV](mpParentProperties,
+                                             rGeometry,
+                                             rDataContainer,
+                                             rCurrentProcessInfo,
+                                             rShapeFunctionValues,
+                                             GaussPointIndex);
+        else
+            return mpParentProperties->GetValue(rV);
+    }
+
+    void Configure(const DoubleVariableType& rV, std::function<double(
+                                                const Properties*,
+                                                const GeometryType&,
+                                                const DataValueContainer&,
+                                                const ProcessInfo&,
+                                                const Vector&,
+                                                const int)> DoubleFunction)
+    {
+        mConfigurationsDouble[rV] = DoubleFunction;
+    }
+
+
+
+    ///@}
+    ///@name Access
+    ///@{
+
+
+    ///@}
+    ///@name Inquiry
+    ///@{
+
+
+    ///@}
+    ///@name Input and output
+    ///@{
+
+    /// Turn back information as a string.
+    std::string Info() const
+    {
+        std::stringstream buffer;
+        buffer << "PropertiesAccessor" ;
```

Comment:
maybe we could output what is being configured here. i mean printing the list of variables for which we provide a configure function

Reply:
good idea, I will see if I can "stringify" the functions